shell/gpu/gpu_surface_gl_delegate.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_GPU_GPU_SURFACE_GL_DELEGATE_H_
+#define FLUTTER_SHELL_GPU_GPU_SURFACE_GL_DELEGATE_H_
+
+#include "flutter/flow/embedded_views.h"
+#include "flutter/fml/macros.h"
+#include "third_party/skia/include/core/SkMatrix.h"
+
+namespace shell {
+
+class GPUSurfaceGLDelegate {
+ public:
+  // Called to make the main GL context current on the current thread.
+  virtual bool GLContextMakeCurrent() = 0;
+
+  // Called to clear the current GL context on the thread. This may be called on
+  // either the GPU or IO threads.
+  virtual bool GLContextClearCurrent() = 0;
+
+  // Called to present the main GL surface. This is only called for the main GL
+  // context and not any of the contexts dedicated for IO.
+  virtual bool GLContextPresent() = 0;
+
+  // The ID of the main window bound framebuffer. Typically FBO0.
+  virtual intptr_t GLContextFBO() const = 0;
+
+  // The rendering subsystem assumes that the ID of the main window bound
+  // framebuffer remains constant throughout. If this assumption in incorrect,
+  // embedders are required to return false from this method. In such cases,
```

Comment:
The name `GLContextFBOResetAfterPresent` sounds to me that returning true requires calling `GLContextFBO()` again after each frame. But the comment seems to suggest otherwise? (return false if `GLContextFBO()` needs to be called)

Reply:
You dont GLContextFBO. Returning try makes the rasterizer ask you for the new FBO (by calling GLContextFBO, which you have overidden).

Reply:
So returning true does mean that someone has to call our GLContextFBO after each frame, right?

Reply:
GPSurfaceGL will automatically call it. There is nothing extra to do. https://github.com/flutter/engine/blob/7c702404b820c670cda34b564ac3f8c0b914f40e/shell/gpu/gpu_surface_gl.cc#L331

Reply:
I understand now. Good catch. Fixed.